src/Sentry/DynamicSamplingContext.cs

```diff
@@ -184,18 +210,20 @@ public static DynamicSamplingContext CreateFromPropagationContext(SentryPropagat
             publicKey,
             null,
             release: release,
-            environment: environment);
+            environment: environment,
+            replaySession: replaySession
+            );
     }
 }
 
 internal static class DynamicSamplingContextExtensions
 {
-    public static DynamicSamplingContext? CreateDynamicSamplingContext(this BaggageHeader baggage)
-        => DynamicSamplingContext.CreateFromBaggageHeader(baggage);
+    public static DynamicSamplingContext? CreateDynamicSamplingContext(this BaggageHeader baggage, IReplaySession? replaySession = null)
+        => DynamicSamplingContext.CreateFromBaggageHeader(baggage, replaySession);
 
-    public static DynamicSamplingContext CreateDynamicSamplingContext(this TransactionTracer transaction, SentryOptions options)
-        => DynamicSamplingContext.CreateFromTransaction(transaction, options);
+    public static DynamicSamplingContext CreateDynamicSamplingContext(this TransactionTracer transaction, SentryOptions options, IReplaySession? replaySession)
+        => DynamicSamplingContext.CreateFromTransaction(transaction, options, replaySession);
 
-    public static DynamicSamplingContext CreateDynamicSamplingContext(this SentryPropagationContext propagationContext, SentryOptions options)
-        => DynamicSamplingContext.CreateFromPropagationContext(propagationContext, options);
+    public static DynamicSamplingContext CreateDynamicSamplingContext(this SentryPropagationContext propagationContext, SentryOptions options, IReplaySession? replaySession)
+        => DynamicSamplingContext.CreateFromPropagationContext(propagationContext, options, replaySession);
```

Comment:
Instead of adding replay to all the factories, how about we use just the `WithReplayId()`?

Reply:
for reference/example, in dart, updating from scope is done like this: https://github.com/getsentry/sentry-dart/blob/e401af3bfe7268f6ad94bbf592fc81525eda264b/dart/lib/src/sentry_baggage.dart#L96-L113

Reply:
I'd rather avoid `WithReplayId` where possible as it has to recreate the `DynamicSamplingContext` in order to add the replay id... that would result in us creating the `DynamicSamplingContext` twice for each of the factories if there was an active replay session 😞 

The other option would be to change `DynamicSamplingContext.Items` from a read only dictionary to a mutable dictionary. Most of the stuff in there really shouldn't be messed with once it's created (we're supposed to be propagating stuff verbatim), so until now it's been nice to have this be immutable. 

All of this stuff is internal, so we can easily refactor in a separate PR whenever we like, if we want to go with a different coding style.